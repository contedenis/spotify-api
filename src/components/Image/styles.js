// @packages
import styled from 'styled-components';

const ImageStyled = styled.img`
  border-radius: ${(props) => (props.type === 'circle' ? '50%' : '5px')};
  border: none;
  height: 200px;
  width: 200px;
`;

export {
  ImageStyled,
};
