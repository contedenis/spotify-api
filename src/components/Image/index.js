// @packages
import React from 'react';
import PropTypes from 'prop-types';

// @own
import { ImageStyled } from './styles';

function Image({
  alt,
  className,
  size,
  src,
  type,
}) {
  return (
    <ImageStyled
      alt={alt}
      className={className}
      size={size}
      src={src}
      type={type}
    />
  );
}

Image.defaultProps = {
  alt: '-image',
  className: '',
  size: 50,
  type: null,
};

Image.propTypes = {
  alt: PropTypes.string,
  className: PropTypes.string,
  size: PropTypes.number,
  src: PropTypes.string.isRequired,
  type: PropTypes.string,
};

export default Image;
