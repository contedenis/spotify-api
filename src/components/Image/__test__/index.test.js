// @packages
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

// @own
import Image from '../index';

describe('Image', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <Image />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('render Image', () => {
    render(<Image alt="image-test" />);

    const chipItem = screen.getByRole('img', { name: 'image-test' });
    expect(chipItem).toBeInTheDocument();
  });
});
