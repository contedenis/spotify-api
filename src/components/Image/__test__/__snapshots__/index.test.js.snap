// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image matches snapshot 1`] = `
<img
  alt="-image"
  class="sc-AxjAm fgYNKY"
  size="50"
/>
`;
